File: BankingAPI/Models/Transaction.cs
namespace BankingAPI.Models
{
    public class Transaction
    {
        public int TransactionId { get; set; }
        public int ClientId { get; set; }
        public string ClientName { get; set; } = string.Empty;
        public string AccountNumber { get; set; } = string.Empty;
        public string TransactionType { get; set; } = string.Empty;
        public decimal TransactionAmount { get; set; }
        public decimal BalanceAfterTransaction { get; set; }
        public DateTime TransactionDate { get; set; }
        public string? Description { get; set; }
    }
}

File: BankingClient/src/app/models/Transaction.ts
export interface Transaction {
  transactionId: number;
  clientId: number;
  clientName: string;
  accountNumber: string;
  transactionType: string;
  transactionAmount: number;
  balanceAfterTransaction: number;
  transactionDate: Date;
  description?: string;
}

export interface CreateTransactionRequest {
  clientId: number;
  transactionType: string;
  transactionAmount: number;
  description?: string;
}
